File: Backend-node/src/routes/albumRoutes.js
const express = require('express');
const router = express.Router();
const albumController = require('../controllers/albumController');
const imageController = require('../controllers/imageController');

router.post('/albums/create', albumController.createAlbum);
router.get('/users/:id_user/albums', albumController.getAlbumByUserId);
router.put('/albums/update/:id_album', albumController.updateAlbum);
router.delete('/albums/delete/:id_album', albumController.deleteAlbum);

router.post('/albums/image/create', imageController.createImage);
router.get('/albums/:id_album/images', imageController.getImageByAlbumId);
router.get('/albums/image/:id_image', imageController.getImageById);

module.exports = router;

File: Backend-python/src/routes/albumRoutes.py
from flask import Blueprint, request, jsonify
import controllers.albumController as albumController
import controllers.imageController as imageController

albumController_bp = Blueprint('albumController_bp', __name__, url_prefix = '/api')

@albumController_bp.route('/albums/create', methods=['POST'])
def createAlbum():
    return albumController.createAlbum()

@albumController_bp.route('/users/<int:id_user>/albums', methods=['GET'])
def getAlbumByUserId(id_user):
    return albumController.getAlbumByUserId(id_user)

@albumController_bp.route('/albums/update/<int:id_album>', methods=['PUT'])
def updateAlbum(id_album):
    return albumController.updateAlbum(id_album)

@albumController_bp.route('/albums/delete/<int:id_album>', methods=['DELETE'])
def deleteAlbum(id_album):
    return albumController.deleteAlbum(id_album)

@albumController_bp.route('/albums/image/create', methods=['POST'])
def createImage():
    return imageController.createImage()

@albumController_bp.route('/albums/<int:id_album>/images', methods=['GET'])
def getImageByAlbumId(id_album):
    return imageController.getImageByAlbumId(id_album)

@albumController_bp.route('/albums/image/<int:id_image>', methods=['GET'])
def getImageById(id_image):
    return imageController.getImageById(id_image)
    